### ISSUE TITLE
Adding actions to show or list views are not accessible!

### ISSUE THREAD

**Issue by TTimmers** [dpgaspar/Flask-AppBuilder#188]
When I add an action to any of my show or list views, when I click them I get a Access Denied flash. 

I am logged in as admin, so it shouldn’t be a rights issue .... I checked permissions, and do see my actions added to the Admin role. I have no idea why this is happening, therefore any help on this would be much appreciated.

**Comment by dpgaspar** [comment]
If you go to the edit role form (for the Admin role record) on the "permissions" field, see if there is any extra permission missing on the multiple drop down.

**Comment by TTimmers** [comment]
I checked that:
- I have a JobModelView, having 2 actions (Kill and Stop);
- the edit form of the Admin Role, contains 2 permissions related to these actions (job kill on JobModelView + job stop on JobModelView)
- these 2 actions, dont do anything (yet) ... other then flash a message.

So as far as I know, this is configured properly. Anything else I can check/try? I probably do something wrong, but it has worked in the past ... although that was with a other version of Flask AppBuilder.

**Comment by dpgaspar** [comment]
ok, post the following:

- Flask-Appbuilder version.
- ModelView code.
- Model Code.

**Comment by TTimmers** [comment]
Thanks for your fast replies!

The info you requested, see below, please let me know when you need more info. 


- Flask-Appbuilder version: 1.4.2
- ModelView code:
```python
class JobModelView(ModelView):
    datamodel = SQLAInterface(Job)

    list_columns = ['name','start_time','PAYLOAD']
    #add_columns = ['name', 'PAYLOAD']
    add_columns = []
    show_fieldsets = [
		('Summary',{'fields':['name','PAYLOAD']}),
		('Job Info',{'fields':['module_type', 'start_time','SRVHOST','SRVPORT','LHOST','LPORT'],'expanded':False}),
		]

    @action("job_kill", "Kill", "Kill all selected?", "fa-rocket", single=False)
    def kill(self, items):
        print items
        for item in items:
            result = "test result"
            flash(str(item.id), "info")
            flash(str(result), "info")
        self.update_redirect()
        return redirect(self.get_redirect())

    @action("job_stop", "Stop", "Stop all selected?", "fa-rocket", single=False)
    def stop(self, items):
        print "stopping"
        return redirect(self.get_redirect())
```

- Model code:
```python
class Job(Model, AuditMixin):
	id = Column(Integer, primary_key=True)
	jid = Column(Integer,  nullable=True)
	name =	Column(String(564), nullable=False)
	module_type = Column(String(150))
	start_time 	= Column(String(150))
	SSL = Column(Boolean)
	SSLVersion = Column(String(20), default='Empty')
	SRVHOST = Column(String(564), default='Empty')
	SRVPORT = Column(BigInteger)
	PAYLOAD = Column(String(564), default='Empty')
	LHOST = Column(String(564), default='Empty')
	LPORT = Column(BigInteger)

	def __repr__(self):
		return self.name
```

**Comment by TTimmers** [comment]
I am not getting any further... I just don't get it. Any help to get me further would help.

**Comment by dpgaspar** [comment]
This looks like a bug :(

Thank you for reporting! *form.action* field is not being filled, i'll correct it and release a new version.

Take a look at the issue_188 example, you can use action on the show view. The problem is on actions on list views (it uses form submit).

**Comment by TTimmers** [comment]
Too bad it turns out to be a bug ... I like your Flask AppBuilder tool very much, therefore I would like to continue using it for my personal tools ... As I use the action option for quite a few of these personal tools, I would like to see this fixed. So, you are welcome for reporting it. :-)

Regarding issue_188 example, I can't seem to find that in the examples ... Am I missing something? Or did you mean issue_186? (which would not make sense to me, as it is not related to the issue you explained above!)

**Comment by dpgaspar** [comment]
Sorry, for the example you have to go to branch 1.4.3

https://github.com/dpgaspar/Flask-AppBuilder/tree/1.4.3/examples/issue_188

**Comment by dpgaspar** [comment]
This bug is present since 1.4.1 release. You can downgrade to 1.4.0. I'll correct it on 1.4.3 ASAP.

**Comment by TTimmers** [comment]
Thanks a lot, downgrading to version 1.4.0 solved the issue. I will be updating again, when it is fixed in 1.4.3. Thanks for your support!

**Comment by dpgaspar** [comment]
No problem @TTimmers, i've corrected the bug and released 1.4.3. Please update.
